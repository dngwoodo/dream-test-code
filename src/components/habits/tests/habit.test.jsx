import React from 'react';

import { render, screen } from "@testing-library/react";

import userEvent from '@testing-library/user-event';

import renderer from 'react-test-renderer';

import Habits from "../habits";

describe('Habits', () => {
  const HABITS = [
    { id: 1, name: 'Reading', count: 0 },
    { id: 2, name: 'Running', count: 0 },
    { id: 3, name: 'Coding', count: 0 },
  ];
  let handleIncrement;
  let handleDecrement;
  let handleDelete;
  let handleReset;
  

  function renderHabits(habits) {
    return render(
     <Habits 
        habits={habits}
        onIncrement={handleIncrement}
        onDecrement={handleDecrement}
        onDelete={handleDelete}
        onReset={handleReset}
      />
    )
  }

  beforeEach(() => {
    handleIncrement = jest.fn();
    handleDecrement = jest.fn();
    handleDelete = jest.fn();
    handleReset = jest.fn();
  })

  it('renders', () => {
    const component = renderer.create(
      <Habits 
        habits={HABITS}
        onIncrement={handleIncrement}
        onDecrement={handleDecrement}
        onDelete={handleDelete}
        onReset={handleReset}
      />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('listens click event', () => {
    renderHabits(HABITS);

    userEvent.click(screen.getByRole('button', {name: 'Reset All'}));

    expect(handleReset).toBeCalledTimes(1);
  })
});