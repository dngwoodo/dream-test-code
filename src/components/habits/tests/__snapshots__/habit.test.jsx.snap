// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Habits renders 1`] = `
<div
  className="habits"
>
  <form
    className="add-form"
    onSubmit={[Function]}
  >
    <input
      className="add-input"
      placeholder="Habit"
      type="text"
    />
    <button
      className="add-button"
    >
      Add
    </button>
  </form>
  <ul>
    <li
      className="habit"
    >
      <span
        className="habit-name"
      >
        Reading
      </span>
      <span
        className="habit-count"
      >
        0
      </span>
      <button
        className="habit-button habit-increase"
        onClick={[Function]}
      >
        <i
          className="fas fa-plus-square"
        />
      </button>
      <button
        className="habit-button habit-decrease"
        onClick={[Function]}
      >
        <i
          className="fas fa-minus-square"
        />
      </button>
      <button
        className="habit-button habit-delete"
        onClick={[Function]}
      >
        <i
          className="fas fa-trash"
        />
      </button>
    </li>
    <li
      className="habit"
    >
      <span
        className="habit-name"
      >
        Running
      </span>
      <span
        className="habit-count"
      >
        0
      </span>
      <button
        className="habit-button habit-increase"
        onClick={[Function]}
      >
        <i
          className="fas fa-plus-square"
        />
      </button>
      <button
        className="habit-button habit-decrease"
        onClick={[Function]}
      >
        <i
          className="fas fa-minus-square"
        />
      </button>
      <button
        className="habit-button habit-delete"
        onClick={[Function]}
      >
        <i
          className="fas fa-trash"
        />
      </button>
    </li>
    <li
      className="habit"
    >
      <span
        className="habit-name"
      >
        Coding
      </span>
      <span
        className="habit-count"
      >
        0
      </span>
      <button
        className="habit-button habit-increase"
        onClick={[Function]}
      >
        <i
          className="fas fa-plus-square"
        />
      </button>
      <button
        className="habit-button habit-decrease"
        onClick={[Function]}
      >
        <i
          className="fas fa-minus-square"
        />
      </button>
      <button
        className="habit-button habit-delete"
        onClick={[Function]}
      >
        <i
          className="fas fa-trash"
        />
      </button>
    </li>
  </ul>
  <button
    className="habits-reset"
    onClick={[MockFunction]}
  >
    Reset All
  </button>
</div>
`;
